day3/m_partitioner.F90

```diff
@@ -254,8 +289,8 @@ function scatter_f32(this, field, root) result(r)
     )
     call error(ierror /= 0, 'Problem with MPI_Scatter', code = ierror)
 
-    i_start = this%domain_(1) + 1
-    j_start = this%domain_(2) + 1
+    i_start = this%domain_(1)
```

Comment:
This was different in Python since `i_start:i_end` is non inclusive of `i_end`, but it is in Fortran array-slicing notation.